### PR TITLE
Add an na argument to read_lines()

### LINKED ISSUE #479: read_lines probably should have na argument
Since write_lines does, and they should be symmetric.

### PR DESCRIPTION
**Pull request by jimhester**
Fixes #479

### CODE REVIEW

File: R/read_lines.R [hunk 1]
```diff
@@ -15,12 +15,13 @@
 #' read_lines_raw(readr_example("mtcars.csv"))
 read_lines <- function(file, skip = 0, n_max = -1L,
                        locale = default_locale(),
+                       na = "NA",
```
Comment (hadley) [inline R/read_lines.R]: I do wonder if this should default to something other than "NA". Maybe nothing?

Comment (jimhester) [inline R/read_lines.R]: Seems reasonable, I'll change it.

**Comment by jimhester** [comment]
I changed the default to `character()`, which allows any line to pass through (including the empty string) by default. `na = ""` will change empty lines to NA.

**Comment by hadley** [comment]
LGTM

**Pull request merged**